vignettes/workflows.Rmd

```diff
@@ -0,0 +1,292 @@
+
+```{r report_stats}
+report_stats(data)
+```
+
+### Visual quality checks
+
+The `plot_quality` function provides a visual check of the nature of the data, by visualizing the distribution of turn durations, and transition timing.
+
+Transition timing is similar to FTO, but calculated without additional quality checks: transitions are identified when the participant changes from one turn to the next. The transition time is then calculated as the difference between the beginning of the turn of the new participant, and the end of the turn of the previous one.
+
+By default, `plot_quality()` will plot the entire dataset:
+
+```{r}
+plot_quality(data)
+
+```
+
+Quality plots can also be run for a specific source:
+
+```{r}
+plot_quality(data, source = "/dutch2/DVA8K")
+```
+
+A quality plot consists of three separate visualizations, all designed to allow rapid visual inspection and spotting oddities:
+
+1. A density plot of turn durations. This is normally expected to look like a distribution that has a peak around 2000ms (2 seconds) and maximum lengths that do not far exceed 10000ms (10 seconds) (Liesenfeld & Dingemanse 2022). The goal of this plot is to allow eyeballing of oddities like turns of extreme durations or sets of turns with the exact same duration (unlikely in carefully segmented conversational data).
+
+2. A density plot of turn transition times. A plot like this is expected to look like a normal distribution centered around 0-200ms (Stivers et al. 2009). Deviations from this may signal problems in the dataset, for instance due to imprecise or automated annotation methods.
+
+3. A scatterplot of turn transition (x) by turn duration (y). This combines both distributions and is expected to look like a cloud of datapoints that is thickest in the middle region. Any standout patterns (for instance, turns whose duration is equal to their transition time) are indicative of problems in the segmentation or timing data.
+
+Each of the three plots can also be generated separately:
+```r
+
+plot_density(data, colname="duration", title="Turn durations",xlab="duration (ms)")
+
+plot_density(data, colname="FTO", title="Turn transitions (FTO)",xlab="FTO (ms)")
+
+plot_scatter(data, colname_x="FTO",colname_y="duration",title="Turn transitions and durations",xlab="transition (ms)", ylab="duration (ms)")
+
+```
+
+
+## Workflow 2: Plot conversations
+
+Another key use of `talkr` is to visualize conversational patterns. A first way to do so is `geom_turn()`, a ggplot2-compatible geom that visualizes the timing and duration of turns in a conversation.
+
+We can start by simply visualizing some of the conversations in the dataset. Here we sample the first four and plot the first minute of each. We display them together using `facet_wrap()` by `source`.
+
+```{r geom_turn_demon1}
+library(ggplot2)
```

Comment:
because you are loading a library already (consider doing this in a separate chunk maybe?), it is perfectly ok to load `dplyr` here, too. Having very explicit function calls is essential inside a package, but less so in a demo like this.

Reply:
Ah, that helps!

Reply:
fixed in commit https://github.com/elpaco-escience/talkr/commit/2cb511858861beabc727f53dbd32a676d172a6fc